packages/dds/tree/src/test/utils.ts

```diff
@@ -1032,11 +1032,50 @@ export interface EncodingTestData<TDecoded, TEncoded, TContext = void> {
 const assertDeepEqual = (a: unknown, b: unknown): void => assert.deepEqual(a, b);
 
 /**
- * Constructs a basic suite of round-trip tests for all versions of a codec family.
+ * Tracks tested versions and registers an after hook to validate that all supported
+ * versions in the codec family have corresponding tests.
+ */
+const testedVersionsByFamily = new WeakMap<
+	ICodecFamily<unknown, unknown>,
+	Set<FormatVersion>
+>();
+const validateRegistered = new WeakSet<ICodecFamily<unknown, unknown>>();
+
+function registerValidationHook<TDecoded, TContext>(
+	family: ICodecFamily<TDecoded, TContext>,
+	versions: Iterable<FormatVersion>,
+): void {
+	let tested = testedVersionsByFamily.get(family);
+	if (tested === undefined) {
+		tested = new Set<FormatVersion>();
+		testedVersionsByFamily.set(family, tested);
+	}
+	for (const version of versions) {
+		tested.add(version);
+	}
+	if (!validateRegistered.has(family)) {
+		validateRegistered.add(family);
+		after("validate all versions tested", () => {
```

Comment:
It's really sad that this `after` hook approach doesn't lead to a noticeable failure when running the test in VSCode's test explorer UI. Worse case we'll get a failure on CI, so It's better than nothing.